docs/RELEASING.md

```diff
@@ -0,0 +1,19 @@
+# Releasing
+
+## Process
+
+For version v0.x.y:
+
+1. Create an annotated tag
+   > NOTE: To use your GPG signature when pushing the tag, use `git tag -s [...]` instead)
+   - `git tag -a v0.x.y -m v0.x.y`
```

Comment:
show is better than tell, and do is best of all. I've got a `git-tag-and-release` script that I've used for same purposes in the old pre-oss repos that has a lot of good logic. Maybe throw into contrib/ and make this `run ./contrib/git-tag-and-release`?

```sh
#!/usr/bin/env bash

set -o errexit -o nounset -o pipefail

prefix=${1:-}
prefix=${1:+$prefix-}

if [[ $(git symbolic-ref HEAD) != refs/heads/master ]] && [[ -z ${ALLOW_NON_MASTER:-} ]]; then
	echo "Must be on master branch" >&2
	exit 1
fi
if [[ $(git describe --dirty) != $(git describe) ]]; then
	echo "Repo must be in a clean state" >&2
	exit 1
fi

last_tag=$(git describe --abbrev=0)
last_tag_commit=$(git rev-list -n1 "$last_tag")
last_specific_tag=$(git tag --contains="$last_tag_commit" | (grep "${prefix:-v}" || echo "HEAD^"))
last_specific_tag_commit=$(git rev-list -n1 "$last_specific_tag")
if [[ $last_specific_tag_commit == $(git rev-list -n1 HEAD) ]]; then
	echo "No commits since last tag" >&2
	exit 1
fi

count=0
git fetch
for count in {0..9}; do
	t=${prefix}v$(TZ=UTC date +%y.%m.%d."$(printf %02d $count)")
	git tag -a -m "$t" "$t" &>/dev/null && echo "created tag $t" >&2 && exit
done
echo "failed to find available tag in 00..09, rough day?" >&2 && exit 1
```

Reply:
though it doesn't do `-s`

Reply:
I think the idea originally was to have a fully automated release process, with just requiring pushing a tag to be a first step towards that.

I think we should continue that approach rather than revert back to having privileged users run scripts locally, since then we would continually reduce the level of privileges needed by users to create a release, especially as the community grows.

Reply:
The script just helps with creating the tag. Does some basic sanity checking like "is git repo clean", "are there even any commits since that last tag?" and then comes up with the tag based on date, which we probably don't want. In the end you get a tag just like your instruction says.

Reply:
Ah, thanks for the clarification, I'll add adapt as needed and add